docs/docs/methods.md

```diff
@@ -1311,3 +1311,233 @@ Do not add properties which are absent in an object but are present in transform
+
+rootReducer([], 1)      // []
+rootReducer([], 2)      // [2]
+rootReducer([2], 3)     // [2]
+rootReducer([2], 4)     // [2, 4]
+rootReducer([2, 4], 5)  // [2, 4]
+```
+
+## `find`
+
+Finds a value in an `array` based on a predicate and returns it (a search is made from a beginning). Otherwise, returns `undefined`
+
+```js
+import { find } from 'nanoutils'
+
+const isDigit = value => value >= 0 && value < 10
+
+find(isDigit, [10, 20, 30])   // undefined
+find(isDigit, [10, 2, 3])     // 2
+```
+
+## `findIndex`
+
+Finds a value in an `array` based on a predicate and returns an index of it (a search is made from a beginning). Otherwise, returns `-1`
+
+```js
+import { findIndex } from 'nanoutils'
+
+const isDigit = value => value >= 0 && value < 10
+
+findIndex(isDigit, [10, 20, 30])    // -1
+findIndex(isDigit, [10, 2, 3])      // 1
+```
+
+## `findLast`
+
+Finds a value in an `array` based on a predicate and returns it (a search is made from an end). Otherwise, returns `undefined`
+
+```js
+import { findLast } from 'nanoutils'
+
+const isDigit = value => value >= 0 && value < 10
+
+findLast(isDigit, [10, 20, 30])   // undefined
+findLast(isDigit, [10, 2, 3])     // 3
+```
+
+## `findLastIndex`
+
+Finds a value in an `array` based on a predicate and returns an index of it (a search is made from an end). Otherwise, returns `-1`
+
+```js
+import { findLastIndex } from 'nanoutils'
+
+const isDigit = value => value >= 0 && value < 10
+
+findLastIndex(isDigit, [10, 20, 30])    // -1
+findLastIndex(isDigit, [10, 2, 3])      // 2
+```
+
+## `flatten`
+
+Returns an `array` with elements of it and its subarrays
+
+```js
+import { flatten } from 'nanoutils'
+
+var array = [1, 2, [3, [4, [5]], 6], [7], 8]
+
+flatten(array)  // [1, 2, 3, 4, 5, 6, 7, 8]
+```
+
+::: tip
+In spite of `flatten`, `unnest` returns an array with unwrapped elements of a depth `1`
+:::
+
+## `flattenObj`
+
+It converts nested object into a flatten one, all keys become paths to repsective values
+
+```js
+import { flattenObj } from 'nanoutils'
+
+const people = {
+  john: {
+    name: 'John Parker',
+    age: 18
+  }
+}
+flattenObj(people)  //  { 'john.name': 'John Parker', 'john.age': 18 }
+```
+
+::: tip
+Remember that `flattenObj` is applied only to `object`s
+
+To apply it to `array`s have a look at [flatten](#flatten)
+:::
+
+## `flip`
+
+Applies a function to a reversed arguments
+
+```js
+import { flip } from 'nanoutils'
+
+const pushReducer = (arr, v) => {
+  arr.push(v)
+  return arr
+}
+const push = flip(pushReducer)
+push(2, [1])   // [1, 2]
+```
+
+## `forEach`
+
+Applies a function to elements of an array
+
+```js
+import { forEach } from 'nanoutils'
+
+const log = forEach(v => console.log(v))
+
+log([1, 2, 3, 4])   // Shows 1, 2, 3, 4 in the console
+```
+
+::: warning
+It may contain side-effects if a function changes elements of an array
+:::
+
+## `forEachObjIndexed`
+
+Walks over an object with a function which can access keys and values
```

Comment:
“Iterates through object’s key-value pairs”
Also it should be told if proto key-value pairs is itereted too or not

Reply:
Interesting fact about iterate
<img width="737" alt="screen shot 2018-09-12 at 02 11 54" src="https://user-images.githubusercontent.com/2991847/45392697-43167c80-b631-11e8-8a30-e4ae7c6f0c35.png">